monero-wallet-gui.pro

```diff
@@ -246,6 +246,11 @@ linux {
             -Wl,-Bdynamic \
             -lGL
     }
+     
```

Comment:
Is there a reason you added a new `CONFIG(static) {}` block here rather than adding to the existing one above on line 220?

More importantly, the change you are proposing undoes a fix from last february: https://github.com/monero-project/monero-core/pull/478 On that PR, @Jaqueeee said "IIRC this would kill the static build. So please add this on non static build only." in [this comment](https://github.com/monero-project/monero-core/pull/478#discussion_r102642782).

Would be good to find out if something changed and `ldl` now works for static and non-static or if this is only correct for some architectures.

Reply:
Thanks for the ping @jonathancross. Libdl is the dynamic linker. It's only required if you're linking dynamic libraries. The reason for the static build to fail could be that you don't have static versions of all the dependent libraries. Note that a static GUI build also requires a statically built Qt (built from source). Use `./build.sh release`instead of `./buid.sh release-static` if you don't need a static build.

Reply:
If i added above the line 249 i would have got the error i was mentioning in the pool "DSO missing from command line" this error i'm getting only if i compile with release-static, i guess there are some codes that need -ldl for some reason, i'm pretty new in c++ i don't really understand why is working if i add the -ldl just after the line 249. What i was sure about this error was that the -ldl was missing.
I'm using Ubuntu 17.10 86_64 with gcc/c++ 7 and boost 1_64, someone more expert than me maybe can explain what and why is -ldl needed for my specs.

Reply:
Unfortunately I'm also inexperienced with C++  :-)

@Jaqueeee Are you saying that `DSO missing from command line` is expected if Qt was not built from source?  Can we add a check to provide a more meaningful message to users?
TIL that [the order in which libs appear on the commandline is critical](https://stackoverflow.com/questions/19901934/strange-linking-error-dso-missing-from-command-line#19905704) and can cause this error message as well.

@italocoin-project Seems you will need to build Qt if you want to use `release-static`.

Reply:
@jonathancross yes, I've realised that after @Jaqueeee ping, i will compile Qt in the next few  days and i'll let a reply here if that was the cause.

Reply:
So, i've compiled the QT from source and i still get this error! Qt compile from source won't solve the problem in my end! Any other ideas?